students/ZackConnaughton/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,140 @@
+"""
+creates database with customer info and product info
+"""
+
+import csv
+from pymongo import MongoClient
+
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def show_available_products():
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.media
+
+        output_dict = list(database.products.find({'quantity_available': {'$gt':'0'}}, {"_id": 0}))
+
+        print(output_dict)
+        return(output_dict)
+
+def show_rentals(product_id):
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.media
+
+        output_list = list(database.rentals.aggregate([
+            {'$match':
+                {'product_id': product_id}},
+            {'$lookup':
+                {'from': "customers",
+                'localField': "user_id",
+                'foreignField': "user_id",
+                'as': "cust"}},
+            {'$project':
+                {'_id': 0,
+                'user_id': 1,
+                'name': '$cust.name',
+                'address': '$cust.address',
+                'phone_number': '$cust.phone_number',
+                'email': '$cust.email'}}
+
+                ]))
+        return(output_list)
+
+
+def print_mdb_collection(collection_name):
+    for doc in collection_name.find({}):
+        print(doc)
+
+def csv_list(file):
+    return_list = []
+    with open('sample_csv_files/' + file, encoding='utf-8-sig') as csv_file:
+        file_input = csv.DictReader(csv_file)
+        for row in file_input:
+            return_list.append(row)
```

Comment:
Great place for a comprehension! 
```return_list = [row for row in file_input]```
but, if that is what you want to do, you can do ```file_input = list(csv.DictReader(csv_file))```
However, I don't think that is what you really want to do. If you are going to iterate through it, you should just leave it as an iterator. Much better memory management.